coordinator/internal/logic/provertask/batch_prover_task.go

```diff
@@ -61,13 +69,47 @@ func (bp *BatchProverTask) Assign(ctx *gin.Context, getTaskParameter *coordinato
 		return nil, fmt.Errorf("check prover task parameter failed, error:%w", err)
 	}
 
+	hardForkNumber, err := bp.getHardForkNumberByName(getTaskParameter.HardForkName)
+	if err != nil {
+		log.Error("hard fork get empty batch because of the hard fork name don't exist", "fork name", getTaskParameter.HardForkName)
+		return nil, err
+	}
+
+	// if the hard fork number set, rollup relayer must generate the chunk from hard fork number,
+	// so the hard fork chunk's start_block_number must be ForkBlockNumber
+	var startChunkIndex uint64 = 0
+	var endChunkIndex uint64 = math.MaxInt64
+	fromBlockNum, toBlockNum := forks.BlockRange(hardForkNumber, bp.forkHeights)
+	if fromBlockNum != 0 {
+		startChunk, chunkErr := bp.chunkOrm.GetChunkByStartBlockNumber(ctx, fromBlockNum)
+		if chunkErr != nil {
+			log.Error("failed to get fork start chunk failure", "forkName", getTaskParameter.HardForkName, "fromBlockNumber", fromBlockNum, "err", chunkErr)
+			return nil, ErrCoordinatorInternalFailure
+		}
+		if startChunk == nil {
+			return nil, nil
+		}
+		startChunkIndex = startChunk.Index
+	}
+	if toBlockNum != math.MaxUint64 {
+		toChunk, chunkErr := bp.chunkOrm.GetChunkByStartBlockNumber(ctx, toBlockNum)
+		if err != nil {
+			log.Error("failed to get fork start chunk failure", "forkName", getTaskParameter.HardForkName, "fromBlockNumber", fromBlockNum, "err", chunkErr)
+			return nil, ErrCoordinatorInternalFailure
+		}
+		if toChunk != nil {
+			// if toChunk is empty, so the startChunk is the last chunk, don't need change the endChunkIndex of math.MaxInt64
+			endChunkIndex = toChunk.Index
```

Comment:
> so the startChunk is the last chunk

this is not necessarily true, there might be more chunks after startChunk. This doesn't change anything in the code tho, just a remark on the comment here.

Reply:
I also don't get the "so the startChunk is the last chunk" part.

I think what's meant here is: "if toChunk is empty, we don't do anything, so `endChunkIndex` will maintain its initial value of `math.MaxInt64`."

Reply:
for examples:

hard fork number : [ 1, 2]

prover hardForkNUmber : 1

block --> block number 1
chunk ---> chunk index 1 | start_block_number 1 | end_block_number 1
batch --- > batch index 1 | start_chunk_index 1 | end_chunk_index 1

after : `fromBlockNum, toBlockNum := forks.BlockRange(hardForkNumber, bp.forkHeights)`
fromBlockNum := 1
endBlockNum := 2

Then `bp.chunkOrm.GetChunkByStartBlockNumber(ctx, toBlockNum)` to get the chunk, it will be null. 

so, the comment is for this case.

> IIUC, toChunk being nil only indicates that we haven't yet reached the fork boundary

@omerfirmak's comment is more suitable.

Reply:
I'll change the comment.

Reply:
fixed [92f5abe](https://github.com/scroll-tech/scroll/pull/1182/commits/92f5abefb95b917f725bd914e2d015346accaab3)